src/Foundatio/Caching/CacheClientHelpers.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Foundatio.Caching
+{
+    public static class CacheClientHelpers
+    {
+        public static async Task<CacheValue<T>> GetOrAddAsync<T>(this ICacheClient client, string key, Func<T> addFunc,
+            TimeSpan? expiresIn = null)
```

Comment:
I'm on my mobile device and curious what the concurrent collections getoradd signature looks like? Can we also add some argument validation